src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal_BiggerRenderChunk_Common.java

```diff
@@ -0,0 +1,230 @@
+ *  THE SOFTWARE.
+ */
+package cubicchunks.asm.mixin.core.client;
+
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Constant;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyConstant;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.Slice;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import cubicchunks.client.CubeProviderClient;
+import cubicchunks.client.IVisGraph;
+import cubicchunks.client.RenderVariables;
+import cubicchunks.util.Coords;
+import cubicchunks.world.ICubicWorldClient;
+import cubicchunks.world.cube.Cube;
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.multiplayer.WorldClient;
+import net.minecraft.client.renderer.RenderGlobal;
+import net.minecraft.client.renderer.ViewFrustum;
+import net.minecraft.client.renderer.chunk.RenderChunk;
+import net.minecraft.client.renderer.chunk.VisGraph;
+import net.minecraft.client.renderer.culling.ICamera;
+import net.minecraft.client.renderer.entity.RenderManager;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.ClassInheritanceMultiMap;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.storage.ExtendedBlockStorage;
+
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+@Mixin(RenderGlobal.class)
+public abstract class MixinRenderGlobal_BiggerRenderChunk_Common {
+
+    @Nullable private BlockPos position;
+    @Shadow private int renderDistanceChunks;
+    @Shadow private ViewFrustum viewFrustum;
+    @Shadow @Final private Minecraft mc;
+    @Shadow private WorldClient world;
+    @Shadow private int renderEntitiesStartupCounter;
+    @Shadow @Final private RenderManager renderManager;
+    @Shadow private int countEntitiesTotal;
+    @Shadow private int countEntitiesRendered;
+    @Shadow private int countEntitiesHidden;
+    
+    @Shadow
+    private boolean isOutlineActive(Entity entityIn, Entity viewer, ICamera camera) {
+        throw new AbstractMethodError();
+    };
+    
+    @Shadow
+    public abstract void loadRenderers();
+    
+    private int renderChunkSizeBit = -1;
+    
+    ClassInheritanceMultiMap<Entity> mutableEntityMapWrapper = new ClassInheritanceMultiMap<Entity>(Entity.class);
+    @SuppressWarnings("unchecked") ClassInheritanceMultiMap<Entity>[] dummyEntityStorageArray = new ClassInheritanceMultiMap[1];
```

Comment:
Does it actually work? Last time I tried it, mixin didn't allow creating arrays in field initializers.

Reply:
Yes.